### PR TITLE
Olivekl add overview

### PR DESCRIPTION
**Pull request by olivekl**
Add overview.md with "Introducing fuzzing" content taken from build_integration.md

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+layout: default
+parent: ClusterFuzzLite
+title: Overview
+has_children: false
+nav_order: 2
+permalink: /overview/
+---
+# Overview
+
+
```
Comment (jonathanmetzman) [inline docs/overview.md:11]: nit: just remove a blank line here.

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+layout: default
+parent: ClusterFuzzLite
+title: Overview
+has_children: false
+nav_order: 2
+permalink: /overview/
+---
+# Overview
```
Comment (jonathanmetzman) [inline docs/overview.md:9]: We should change clusterfuzzlite.md to link to this.

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+## Introducing fuzzing with libFuzzer and Sanitizers.
+
+This section provides an overview of the fuzzing process and defines common terms. 
+If you are already familiar with the process of fuzzing, feel free to skip to [Build Integration]
```
Comment (jonathanmetzman) [inline docs/overview.md:15]: "process of fuzzing"
isn't as good as saying "libFuzzer and Sanitizers"
People familiar with fuzzing but not libFuzzer and Sanitizers should read this.

File: docs/overview.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+If you are already familiar with the process of fuzzing, feel free to skip to [Build Integration] 
+to begin writing fuzzers and integrating with ClusterFuzzLite's build system.
+
+### Fuzzing
+
+[Fuzzing] is a technique were randomized inputs are automatically created and
+fed as input to a (target) program in order to find bugs in that program.
+The program that creates the inputs is called a fuzzer.
+Fuzzing is highly effective at finding bugs missed by manually written tests,
+code review or auditing.
```
Comment (jonathanmetzman) [inline docs/overview.md:24]: nit: should there be a comma after code review?

File: docs/overview.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+// fuzz_target.cc
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  DoSomethingInterestingWithMyAPI(Data, Size);
+  return 0;  // Non-zero return values are reserved for future use.
+}
+```
+Clang's `-fsanitizer=fuzzer` option will link this fuzz target function against
+libFuzzer, producing a fuzzer binary that will fuzz your target code when run.
+Note that in ClusterFuzzLite, you will not use this flag directly, you should
+use the `$LIB_FUZZING_ENGINE` environment variable which we discuss later in
```
Comment (jonathanmetzman) [inline docs/overview.md:55]: change to "which we discuss later in [build integration]"

File: docs/overview.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+With that, you have enough background to build fuzzers for ClusterFuzzLite.
+
+[Fuzzing]: https://en.wikipedia.org/wiki/Fuzzing
+[LibFuzzer]: https://llvm.org/docs/LibFuzzer.html
+[fuzz target]: https://github.com/google/fuzzing/blob/master/docs/glossary.md#fuzz-target
+[AddressSanitizer (ASan)]: https://clang.llvm.org/docs/AddressSanitizer.html
+[UndefinedBehaviorSanitizer (UBSan)]: https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
+[MemorySanitizer (MSan)]: https://clang.llvm.org/docs/MemorySanitizer.html
+[Build Integration]: {{ site.baseurl }}/build-integration/
+
```
Comment (jonathanmetzman) [inline docs/overview.md:93]: nit: remove this line.

File: docs/overview.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+- [MemorySanitizer (MSan)]: For detecting use of uninitialized memory. MSan
+  is the hardest sanitizer to use because an MSan instrumented binary must be
+  entirely instrumented with MSan. If any part of the binary is not instrumented
+  with MSan, MSan will report false positives.
+
+The ClusterFuzzLite codebase uses shorter names for the sanitizers. So when
+referring to a sanitizer when giving input to ClusterFuzzLite, ASan is
+`address`, UBSan is `ubsan` and MSan is `memory`.
+
+With that, you have enough background to build fuzzers for ClusterFuzzLite.
```
Comment (jonathanmetzman) [inline docs/overview.md:84]: nit add link to `[build integration]` doc.

**Review by jonathanmetzman — CHANGES_REQUESTED** [review]

File: docs/overview.md (hunk 2 above)
Comment (olivekl) [reply docs/overview.md:9]: Done. See related updates on clusterfuzzlite.md as well.

File: docs/clusterfuzzlite.md [hunk 8]
```diff
@@ -14,10 +14,12 @@ ClusterFuzzLite is based on [ClusterFuzz].
 Fuzzing is a highly effective technique for finding bugs in software.
 ClusterFuzzLite makes your code more secure by fuzzing changes before they even
 enter the codebase.
+
 ClusterFuzzLite supports [GitHub Actions] and [Google Cloud Build], so getting
 the benefits of continuous fuzzing is simple if you can use these CI systems.
 Suppport for more CI systems is in-progress and
-[extending support to other CI systems] is easy.
+[extending support to other CI systems] is easy. ClusterFuzzLite supports programming languages and runtimes
```
Comment (jonathanmetzman) [inline docs/clusterfuzzlite.md:21]: Change to: "ClusterFuzzLite supports programming languages and runtimes beyond C and C++, including:

File: docs/overview.md (hunk 2 above)
Comment (jonathanmetzman) [reply docs/overview.md:9]: THanks!

**Comment by jonathanmetzman** [comment]
Just one more change to make. Feel free to merge after.

**Pull request merged**